Source/santad/DataLayer/WatchItems.mm

```diff
@@ -684,37 +684,119 @@ bool ParseConfig(NSDictionary *config, SetSharedDataWatchItemPolicy &data_polici
   return diff;
 }
 
-bool DataWatchItems::Build(SetSharedDataWatchItemPolicy data_policies) {
+void FindMatches(NSString *base, NSMutableArray<NSString *> *path_components, NSUInteger idx,
+                 std::vector<std::string> &matches) {
+  if (path_components.count == idx) {
+    // Nothing left to match, add the current full base path
+    matches.push_back(base.UTF8String);
+    return;
+  }
+
+  NSString *path = [NSString stringWithFormat:@"%@%@", base, path_components[idx]];
+
   glob_t *g = (glob_t *)alloca(sizeof(glob_t));
```

Comment:
Could we not put this on the stack?

Reply:
Unfortunately not. This is an Objective-C++ issue with how the struct is defined.

```
....mm:18:10: error: call to implicitly-deleted default constructor of 'glob_t'
   18 |   glob_t g;
      |          ^
In file included from ....mm:9:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.4.sdk/usr/include/glob.h:63:9:
  note: default constructor of '(anonymous union at /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.4.sdk/usr/include/glob.h:59:2)'
  is implicitly deleted because variant field 'gl_errblk' is an ObjC pointer
   63 |                 int (^gl_errblk)(const char *, int);
```

Reply:
Ack